model/src/main/java/org/openremote/model/teltonika/State.java

```diff
@@ -0,0 +1,298 @@
+package org.openremote.model.teltonika;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validator;
+import org.openremote.model.Constants;
+import org.openremote.model.attribute.Attribute;
+import org.openremote.model.attribute.AttributeMap;
+import org.openremote.model.attribute.MetaItem;
+import org.openremote.model.attribute.MetaMap;
+import org.openremote.model.geo.GeoJSONPoint;
+import org.openremote.model.rules.AssetState;
+import org.openremote.model.util.ValueUtil;
+import org.openremote.model.validation.AssetStateStore;
+import org.openremote.model.value.*;
+
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.*;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.openremote.model.value.MetaItemType.*;
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({
+        "reported"
+})
+public class State {
+
+//    public ReportedState reportedState;
+    @JsonProperty("reported")
+    private Map<String, Object> reported;
+
+
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(State.class.getName()).append('@').append(Integer.toHexString(System.identityHashCode(this))).append('[');
+        sb.append("reported");
+        sb.append('=');
+        sb.append(((this.reported == null) ? "<null>" : this.reported));
+        sb.append(',');
+        if (sb.charAt((sb.length() - 1)) == ',') {
+            sb.setCharAt((sb.length() - 1), ']');
+        } else {
+            sb.append(']');
+        }
+        return sb.toString();
+    }
+
+
+    //TODO: Add real timestamp reported by device, check issue #1156
+    //TODO: Add description of each attribute (No idea where to put that yet)
+    //TODO: Move any function from here; it's a POJO.
+    //TODO: Figure out what the parameters: pr, alt, ang, sat, sp, and evt do
+    // and implement their functionality (I can guess but I want to know exactly).
+    public AttributeMap GetAttributes(Map<Integer, TeltonikaParameter> params, AttributeMap additionalAttributes, Logger logger){
```

Comment:
additionalAttributes doesn't seem to be used?

Reply:
It is part of my rethinking of how I should deal with #1156 , meaning if I should pass all attributes I would like to eventually add/update to the Asset or if I should use the returned AttributeMap to append any attributes needed. How should I continue?